CustomTabBar/TabButton.swift

```diff
@@ -40,6 +40,11 @@ class TabButton: UIView {
             updateInsets()
         }
     }
+    var titleFont: UIFont? = nil {
+        didSet {
+            titleLabel.font = titleFont
```

Comment:
Do the optional checking here too. You don't want to set the font to `nil` if someone decided to remove the override. I would have this:

```
if let titleFont = titleFont {
   titleLabel.font = titleFont
} else if dataObject.imageName != nil {
   titleLabel.font = UIFont.systemFontOfSize(10)
} else {
   titleLabel.font = UIFont.systemFontOfSize(16)
}
```

You could also pull that login into a private `updateFont()` function and call it here and in `setupViews()`. That might be a bit cleaner and more obvious.